### ISSUE TITLE
Remove newtonsoft dependency

### ISSUE THREAD

**Issue by DoubleDBE** [horse-framework/horse-messaging#157]
Hello,

Since .NET provides its own JSON serialization library, how about removing the dependency on Newtonsoft's library?

That way we could make use of [source generation](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation) to improve performance (and avoid reflection for internal messages), and have a dependency less to worry about.

The implementation for Newtonsoft's library could be moved to a separate nuget package for those who need it, or we could provide a code example on how to use `IMessageContentSerializer` since it's not that hard.

I'm more than happy to contribute and create a PR!

**Comment by mhelvacikoylu** [comment]
Hi,

IMessageContentSerializer for direct, queue, channel and router message serialization. In previous versions there was only newtonsoft json. After System.Text.Json got popular we added SystemJsonContentSerializer. But some settings still supports only newtonsoft (server side option files, setting queue options, some clustering node operations etc). Because we used some functionalities of Netwonsoft which System.Text.Json does not support.

But yes, it's possible to remove newtonsoft json serialization and set System.Text.Json as default. But we need change some codes. It's possible to see which code lines should change by removing dependency of newtonsoft.
